base/file.jl

```diff
@@ -48,10 +46,12 @@ julia> pwd()
 ```
 """
 function pwd()
-    b = Vector{UInt8}(undef, 1024)
-    len = RefValue{Csize_t}(length(b))
-    uv_error(:getcwd, ccall(:uv_cwd, Cint, (Ptr{UInt8}, Ptr{Csize_t}), b, len))
-    String(b[1:len[]])
+    path_max = 1024
+    buf = Base.StringVector(path_max)
+    sz = RefValue{Csize_t}(path_max + 1)
+    uv_error(:cwd, ccall(:uv_cwd, Cint, (Ptr{UInt8}, Ptr{Csize_t}), buf, sz))
```

Comment:
This function also doesn't handle the case when the current path is larger than `path_max` , should use a similar logic to that used in `homedir`

Reply:
Unfortunately `uv_cwd` doesn't seem to produce ENOBUFS errors: https://github.com/JuliaLang/libuv/blob/26dbe5672c33fc885462c509fe2a9b36f35866fd/src/unix/core.c#L668-L682

… I take that back, the Windows implementation does return ENOBUFS (probably because the encoding difference means that there is no simple way to provide a maximum UTF-8 path): https://github.com/JuliaLang/libuv/blob/26dbe5672c33fc885462c509fe2a9b36f35866fd/src/win/util.c#L202 … so it should be handled.

Reply:
I just checked the docs http://docs.libuv.org/en/v1.x/misc.html#c.uv_cwd

Reply:
I'm not sure why libuv doesn't check if `ERANGE` is thrown and produce `ENOBUFS` errors. It's certainly possible to run into this if `path_max` is too small.

Reply:
Apparently this is just a libuv bug that is fixed https://github.com/libuv/libuv/pull/2335